pkg/server/plugin/nodeattestor/k8s/psat/psat_test.go

```diff
@@ -127,62 +127,40 @@ func (s *AttestorSuite) SetupSuite() {
 
 func (s *AttestorSuite) SetupTest() {
 	s.mockCtrl = gomock.NewController(s.T())
-	s.attestor = s.configureAttestor()
+	s.attestor = s.loadPlugin()
 }
 
-func (s *AttestorSuite) TeardownTest() {
+func (s *AttestorSuite) TearDownTest() {
 	s.mockCtrl.Finish()
 }
 
 func (s *AttestorSuite) TestAttestFailsWhenNotConfigured() {
-	resp, err := s.doAttestOnAttestor(s.newAttestor(), &nodeattestorv0.AttestRequest{})
-	s.RequireGRPCStatus(err, codes.Unknown, "k8s-psat: not configured")
-	s.Require().Nil(resp)
+	attestor := new(nodeattestor.V1)
+	plugintest.Load(s.T(), BuiltIn(), attestor)
+	s.attestor = attestor
+	s.requireAttestError([]byte("{"), codes.FailedPrecondition, "nodeattestor(k8s_psat): not configured")
 }
 
-func (s *AttestorSuite) TestAttestFailsWithNoAttestationData() {
-	s.requireAttestError(&nodeattestorv0.AttestRequest{},
-		"k8s-psat: missing attestation data")
+func (s *AttestorSuite) TestAttestFailsWithMalformedPayload() {
+	s.requireAttestError([]byte("{"), codes.InvalidArgument, "nodeattestor(k8s_psat): failed to unmarshal data payload")
 }
 
-func (s *AttestorSuite) TestAttestFailsWithWrongAttestationDataType() {
-	s.requireAttestError(&nodeattestorv0.AttestRequest{
-		AttestationData: &common.AttestationData{
-			Type: "blah",
-		},
-	}, `k8s-psat: unexpected attestation data type "blah"`)
-}
-
-func (s *AttestorSuite) TestAttestFailsWithNoAttestationDataPayload() {
```

Comment:
We are losing coverage for this case.

Reply:
This case isn't actually easy to test anymore, since the facade now prevents sending an empty payload.